src/precice/precice.hpp

```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+#include <precice/Participant.hpp>
+#include <precice/Version.h>
+#include <precice/span.hpp>
```

Comment:
Isn't this primarily an implementation detail of `Participant.hpp` similar to `export.hpp` and `types.hpp`?
If so, then this doesn't really belong here.

Reply:
But then, would the `precice.hpp` just be a redirection/alias for `Participant.hpp`?

Reply:
I thought we wanted to expose the `Version.h` and `span.hpp` to the user, to be able to use, e.g., the preCICE span directly in their code.

We can achieve this with either location, but I thought that putting them directly into the `precice.hpp` makes the intention clearer.

Reply:
Key point here is that the user may have to create a `precice::span` in the adapter.
So, let's keep it in the `precice.hpp`.

Reply:
And same for `Version.h`, right? So, everything right, conversation resolved, and PR approved?